<?xml version="1.0"?>

<launch>
    <arg name="bag" default="$(find frontier_exploration)/data/josef_2019-06-06-13-58-12_proc_0.1m.bag"/>
    <arg name="bag_rate" default="10.0"/>
    <arg name="play_bag" default="false"/>
    <arg name="rviz_config" default="devel"/>

    <param name="use_sim_time" value="true"/>

    <node name="pc_processor" pkg="frontier_exploration" type="pc_processor.py" output="screen">
        <param name="pointcloud_topic" value="/final_cost_cloud"/>
        <param name="min_dist" value="1.0"/>
        <param name="max_dist" value="10.0"/>
    </node>

    <node if="$(arg play_bag)" name="play" pkg="rosbag" type="play" output="screen"
          args="-r $(arg bag_rate) -k $(arg bag)"/>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find frontier_exploration)/config/$(arg rviz_config).rviz"/>
    <node if="0" name="rqt_tf_tree" pkg="rqt_tf_tree" type="rqt_tf_tree"/>
</launch>
